include/zephyr/dt-bindings/reserved-memory/nordic-owned-memory.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_RESERVED_MEMORY_NORDIC_OWNED_MEMORY_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_RESERVED_MEMORY_NORDIC_OWNED_MEMORY_H_
+
+#include <zephyr/sys/util_macro.h>
+
+/** @cond INTERNAL_HIDDEN */
+#define PREPEND(var, prefix) prefix ## var
+/** @endcond */
+
+/**
+ * @name Memory permission flags.
+ * @{
+ */
+
+/** Readable. */
+#define NRF_PERM_READ BIT(0)
+/** Writable. */
+#define NRF_PERM_WRITE BIT(1)
+/** Executable. */
+#define NRF_PERM_EXEC BIT(2)
+/** Secure-only. */
+#define NRF_PERM_SECURE BIT(3)
+/** Non-secure-callable. */
+#define NRF_PERM_NSC BIT(4)
+
+/**
+ * }@
+ */
+
+/**
+ * @brief Utility macro to build a permission bitfield for a nordic,access property.
+ *
+ * @param ... List of permissions, such as READ, named after NRF_PERM_* definitions.
+ */
+#define NRF_PERM(...) (FOR_EACH_FIXED_ARG(PREPEND, (|), NRF_PERM_, __VA_ARGS__))
```

Comment:
not a fan of macrology in DT, we could just tell users to use eg `NRF_PERM_READ | NRF_PERM_WRITE`.